projects/components/src/stepper/stepper.component.ts

```diff
@@ -0,0 +1,140 @@
+            </ng-template>
+            <ng-container *ngTemplateOutlet="step.content"></ng-container>
+            <ng-template matStepperIcon="edit">
+              <ht-icon [icon]="step.icon" size="${IconSize.ExtraSmall}"> </ht-icon>
+            </ng-template>
+          </mat-step>
+        </ng-container>
+      </mat-stepper>
+      <div class="action-buttons">
+        <ht-button
+          class="cancel"
+          label="Cancel"
+          (click)="this.onClickCancel()"
+          role="${ButtonRole.Tertiary}"
+        ></ht-button>
+        <div class="back-next">
+          <ht-button
+            class="back"
+            *ngIf="stepper.selectedIndex !== 0"
+            display="${ButtonStyle.Outlined}"
+            label="Back"
+            (click)="this.onClickBack(stepper)"
+          ></ht-button>
+          <ht-button
+            class="next"
+            *ngIf="stepper.selectedIndex !== steps.length - 1"
+            label="Next"
+            (click)="this.onClickNext(stepper)"
+            [disabled]="
+              this.isNextDisabled
+                | htMemoize
+                  : this.isLinear
+                  : stepper.steps.get(stepper.selectedIndex)?.completed
+                  : stepper.steps.get(stepper.selectedIndex)?.stepControl?.valid
+            "
+          ></ht-button>
+          <ht-button
+            class="submit"
+            *ngIf="stepper.selectedIndex === steps.length - 1"
+            label="Submit"
+            (click)="this.onClickSubmit()"
+            [disabled]="
+              this.isSubmitDisabled
+                | htMemoize
+                  : stepper.steps.get(stepper.selectedIndex)?.completed
+                  : stepper.steps.get(stepper.selectedIndex)?.stepControl?.valid
+            "
+          ></ht-button>
+        </div>
+      </div>
+    </div>
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  styleUrls: ['./stepper.component.scss']
+})
+export class StepperComponent implements AfterContentInit {
+  // TODO: Fix support for custom icon, currently broken.
+
+  @ContentChildren(StepperTabComponent)
+  private readonly steps!: QueryList<StepperTabComponent>;
+
+  // If stepper is Linear, then we can navigate to a tab only if previous all tabs are at complete state.
+  @Input()
+  public isLinear: boolean = false;
+
+  // Currently we only support horizontal orientation.
+  @Input()
+  public orientation: StepperOrientation = StepperOrientation.Horizontal;
+
+  // True for submit, false for cancel.
+  @Output()
+  public readonly submitForm: EventEmitter<boolean> = new EventEmitter();
+
+  public steps$?: Observable<StepperTabComponent[]>;
+
+  public ngAfterContentInit(): void {
+    this.steps$ = queryListAndChanges$(this.steps).pipe(map(list => list.toArray()));
+  }
+
+  public onClickNext(stepper: MatStepper): void {
+    stepper.next();
+  }
+
+  public onClickBack(stepper: MatStepper): void {
+    stepper.previous();
+  }
```

Comment:
do we really need these functions, we can do it directly in the template, right?

Reply:
Agreed. Moved it to template.